clang/lib/Sema/SemaOverload.cpp

```diff
@@ -14310,8 +14310,8 @@ Sema::CreateOverloadedUnaryOp(SourceLocation OpLoc, UnaryOperatorKind Opc,
         PartialDiagnosticAt(OpLoc, PDiag(diag::err_ovl_deleted_oper)
                                        << UnaryOperator::getOpcodeStr(Opc)
                                        << Input->getSourceRange()),
-        *this, OCD_AllCandidates, ArgsArray, UnaryOperator::getOpcodeStr(Opc),
-        OpLoc);
+        *this, OCD_AllCandidates, ArgsArray.slice(1),
```

Comment:
Can you explain why `ArgArray.slice(1)` fixes the problem? It is not obvious to me from looking at `NoteCandidates`

Reply:
The problem `ArgArray.slice(1)` solves is that `CreateOverloadedUnaryOp` fills first element `ArgsArray` with the object whose method was called. Later in `NoteCandidates` size of `ArgsArray` is passed further and it eventually ends up compared to number of function candidate parameters which never includes implicit object parameter.

Reply:
Can you add that as a comment so that future us don't get confused?  :)

Reply:
> Can you add that as a comment so that future us don't get confused?

Sure, done.

Reply:
> IMO, ArgsArray.drop_front() makes more sense to me here. 

Okay, done.